File: CS-Examples/05_Paragraph/HideParagraph.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Spire.Doc;
using Spire.Doc.Documents;
using Spire.Doc.Fields;

namespace HideParagraph
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            // Create a new Document object.
            Document document = new Document();

            // Load an existing Word document from the specified file path.
            document.LoadFromFile(@"..\..\..\..\..\..\Data\Template_Docx_1.docx");

            // Get the first Section of the document.
            Section sec = document.Sections[0];

            // Get the first Paragraph of the Section.
            Paragraph para = sec.Paragraphs[0];

            // Iterate over the child DocumentObjects in the Paragraph.
            foreach (DocumentObject obj in para.ChildObjects)
            {
                // Check if the child DocumentObject is a TextRange.
                if (obj is TextRange)
                {
                    // Convert the child DocumentObject to a TextRange.
                    TextRange range = obj as TextRange;

                    // Set the Hidden property of the TextRange's CharacterFormat to true, hiding the text.
                    range.CharacterFormat.Hidden = true;
                }
            }

            // Specify the file name for the resulting Word document.
            string result = "Result-HideWordParagraph.docx";

            // Save the Document object to a file in Docx format with compatibility mode set to Docx2013.
            document.SaveToFile(result, FileFormat.Docx2013);

            // Dispose the Document object.
            document.Dispose();

            //Launch the MS Word file.
            WordDocViewer(result);
        }

        private void WordDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }
    }
}

File: VB-Examples/05_Paragraph/HideParagraph.vb
Imports System.ComponentModel
Imports System.Text
Imports Spire.Doc
Imports Spire.Doc.Documents
Imports Spire.Doc.Fields

Namespace HideParagraph
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			' Create a new Document object
			Dim document As New Document()

			' Load the document from the specified file
			document.LoadFromFile("..\..\..\..\..\..\Data\Template_Docx_1.docx")

			' Get the first Section in the document
			Dim sec As Section = document.Sections(0)

			' Get the first Paragraph in the section
			Dim para As Paragraph = sec.Paragraphs(0)

			' Iterate through each ChildObject in the paragraph
			For Each obj As DocumentObject In para.ChildObjects

				' Check if the current ChildObject is a TextRange
				If TypeOf obj Is TextRange Then
				
					' Cast the ChildObject to a TextRange
					Dim range As TextRange = TryCast(obj, TextRange)
					
					' Set the Hidden property of the TextRange to True (hide the text)
					range.CharacterFormat.Hidden = True
					
				End If
				
			Next obj

			' Specify the filename for the output result
			Dim result As String = "Result-HideWordParagraph.docx"

			' Save the modified document to the specified file in Docx2013 format
			document.SaveToFile(result, FileFormat.Docx2013)

			' Dispose the Document object to release resources
			document.Dispose()

			'Launch the MS Word file.
			WordDocViewer(result)
		End Sub

		Private Sub WordDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub
	End Class
End Namespace
